tests/fixture_valid.js

```diff
@@ -246,14 +177,165 @@ module.exports.checkFixture = function checkFixture(fixture, usedShortNames=[])
+        result.errors.push({
+          description: `shortName '${modeShortName}' not unique in mode #${i}.`,
+          error: null
+        });
+      }
+      usedModeShortNames.push(modeShortName);
+
+      if (/\bmode\b/i.test(mode.name) || /\bmode\b/i.test(mode.shortName)) {
+        result.errors.push({
+          description: `mode name and shortName must not contain the word 'mode' in mode '${modeShortName}' (#${i}).`,
+          error: null
+        });
+      }
+
+      if ('physical' in mode
+        && 'lens' in mode.physical
+        && 'degreesMinMax' in mode.physical.lens
+        && mode.physical.lens.degreesMinMax[0] > mode.physical.lens.degreesMinMax[1]
+        ) {
+        result.errors.push({
+          description: `physical.lens.degreesMinMax is an invalid range in mode '${modeShortName}' (#${i}).`,
+          error: null
+        });
+      }
+
+      for (let chIndex = 1; chIndex <= mode.channels.length; chIndex++) {
+        const ch = mode.channels[chIndex-1];
+
+        if (ch === null) {
+          continue;
+        }
+
+        usedChannels.push(ch);
+
+        // it can be a normal channel
+        if (ch in fixture.availableChannels) {
+          if (fixture.availableChannels[ch].type === 'Pan') {
+            checkPanTiltMaxExistence(result, fixture, mode, ch, 'panMax');
+          }
+          else if (fixture.availableChannels[ch].type === 'Tilt') {
+            checkPanTiltMaxExistence(result, fixture, mode, ch, 'tiltMax');
+          }
+          continue;
+        }
+
+        // or it is a fine channel
+        if (ch in fineChannels) {
+          if (!mode.channels.includes(fineChannels[ch])) {
+            result.errors.push({
+              description: `mode '${modeShortName}' uses switching channel '${ch}' (#${chIndex}) but is missing its trigger channel '${fineChannels[ch]}'`,
+              error: null
+            });
+          }
+          continue;
+        }
+
+        // or it is a switching channel
+        if (ch in switchingChannels) {
+          // the mode must also contain the trigger channel
+          if (!mode.channels.includes(switchingChannels[ch])) {
+            result.errors.push({
+              description: `mode '${modeShortName}' uses switching channel '${ch}' (#${chIndex}) but is missing its trigger channel '${switchingChannels[ch]}'`,
+              error: null
+            });
+          }
+          continue;
+        }
+
+        // the channel doesn't exist
+        result.errors.push({
+          description: `channel '${ch}' (#${chIndex}) referenced from mode '${modeShortName}' but is not defined. Note: fine channels can only be used in the same mode as their coarse counterpart.`,
+          error: null
+        });
+      }
+    }
+
+    if ('heads' in fixture) {
+      for (const key in fixture.heads) {
+        const head = fixture.heads[key];
+
+        for (let i = 0; i < head.length; i++) {
+          if (!fixture.availableChannels[head[i]]) {
```

Comment:
Definitely yes. (Even though I don't think that I actively removed this functionality)